src/components/Dialogs/Conditions/ConditionsManager.js

```diff
@@ -162,42 +175,95 @@ const ConditionsManager = ({
                     </div>
                 ))
             }
+            case DIMENSION_TYPE_PHONE_NUMBER: {
+                return conditionsList.map((condition, index) => (
+                    <div key={index}>
+                        <AlphanumericCondition
+                            condition={condition}
+                            onChange={(value) => setCondition(index, value)}
+                            onRemove={() => removeCondition(index)}
+                        />
+                        {getDividerContent(index)}
+                    </div>
+                ))
+            }
             case DIMENSION_TYPE_TEXT:
             case DIMENSION_TYPE_LONG_TEXT:
             case DIMENSION_TYPE_LETTER:
-            case DIMENSION_TYPE_PHONE_NUMBER:
             case DIMENSION_TYPE_EMAIL:
             case DIMENSION_TYPE_USERNAME:
             case DIMENSION_TYPE_URL: {
-                const allowCaseSensitive =
-                    dimensionType !== DIMENSION_TYPE_PHONE_NUMBER
                 return conditionsList.map((condition, index) => (
                     <div key={index}>
-                        <AlphanumericCondition
+                        <CaseSensitiveAlphanumericCondition
                             condition={condition}
                             onChange={(value) => setCondition(index, value)}
                             onRemove={() => removeCondition(index)}
-                            allowCaseSensitive={allowCaseSensitive}
                         />
                         {getDividerContent(index)}
                     </div>
                 ))
             }
-            case DIMENSION_TYPE_BOOLEAN:
-            case DIMENSION_TYPE_TRUE_ONLY: {
-                const showFalseOption = dimensionType === DIMENSION_TYPE_BOOLEAN
+            case DIMENSION_TYPE_BOOLEAN: {
                 return (conditionsList.length && conditionsList).map(
                     (condition, index) => (
                         <div key={index}>
                             <BooleanCondition
                                 condition={condition}
                                 onChange={(value) => setCondition(index, value)}
-                                showFalseOption={showFalseOption}
                             />
                         </div>
                     )
                 )
             }
+            case DIMENSION_TYPE_TRUE_ONLY: {
+                return (conditionsList.length && conditionsList).map(
+                    (condition, index) => (
+                        <div key={index}>
+                            <TrueOnlyCondition
+                                condition={condition}
+                                onChange={(value) => setCondition(index, value)}
+                            />
+                        </div>
+                    )
+                )
+            }
+            case DIMENSION_TYPE_DATE: {
+                return conditionsList.map((condition, index) => (
```

Comment:
Should all check for `conditionsList.length` like the one above?

Reply:
Actually, that's a copy paste error. Neither need to check for `conditionsList.length`, that was used for the numeric types to switch between either `conditionsList` or `legendSet`. Remove the redundant checks